lib/theme_check/shopify_liquid/tag.rb

```diff
@@ -8,6 +8,38 @@ module Tag
 
       def labels
         @tags ||= YAML.load(File.read("#{__dir__}/../../../data/shopify_liquid/tags.yml"))
+          .to_set
+      end
+
+      # Add magic regex methods for all the liquid tags
+      # - if_tag => /\A\{\%-?\s*if/
+      # - if_liquid_tag => /^\s*if/m
+      def method_missing(method)
```

Comment:
Why not a plain method? `find_tag`?
`method_missing` can lead to confusion.

Reply:
I wrote it this way to reduce noise in the code that calls it + make sure all the regexes are following the same pattern (supports both `{%` and `{%-`).

It's already pretty verbose:

```
is_if_tag = markup =~ ShopifyLiquid::Tag.if_tag
is_if_liquid_tag = is_liquid_tag && last_line =~ ShopifyLiquid::Tag.if_liquid_tag
```

But it's true that this might not be much different from

```
is_if_tag = markup =~ ShopifyLiquid::Tag.tag_regex('if')
is_if_liquid_tag = is_liquid_tag && last_line =~ ShopifyLiquid::Tag.liquig_tag_regex('if')
```

I'll change it. I guess I wanted to play with those metaprogramming chops I learned from Why's poignant guide :p